File: app/src/main/java/com/ddu/sdk/network/OkHttpUtils.java
package com.ddu.sdk.network;

import java.util.Map;

import okhttp3.HttpUrl;

/**
 * Created by yzbzz on 2018/1/17.
 */

public class OkHttpUtils {
    
    public static String appendUrl(String url, Map<String, String> urlParams) {
        if (null == url) {
            return "";
        }
        if (urlParams == null || urlParams.size() <= 0) {
            return url;
        }
        HttpUrl httpUrl = HttpUrl.parse(url);
        if (null == httpUrl) {
            return url;
        }
        HttpUrl.Builder urlBuilder = httpUrl.newBuilder();
        for (Map.Entry<String, String> entry : urlParams.entrySet()) {
            urlBuilder.addQueryParameter(entry.getKey(), entry.getValue());
        }
        return urlBuilder.build().toString();
    }
}

File: app/src/main/java/com/ddu/sdk/network/OkHttpUtils.kt
package com.ddu.sdk.network

import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull

/**
 * Created by yzbzz on 2018/2/26.
 */
object OkHttpUtils {

    fun appendUrl(url: String, urlParams: HashMap<String, String>): String {
        if (urlParams.size <= 0) {
            return url
        }
        val httpUrl: HttpUrl? = url.toHttpUrlOrNull()
        if (null == httpUrl) {
            return url
        }
        val urlBuilder = httpUrl?.newBuilder()
        for ((key, value) in urlParams.entries) {
            urlBuilder.addQueryParameter(key, value)
        }
        return urlBuilder.build().toString()
    }
}